integreat_cms/cms/forms/events/event_filter_form.py

```diff
@@ -148,6 +158,17 @@ def apply(
         elif recurrence.NOT_RECURRING in self.cleaned_data["recurring"]:
             # Only non-recurring events
             events = events.filter(recurrence_rule__isnull=True)
+
+        if (
+            calendar.EVENT_NOT_FROM_EXTERNAL_CALENDAR
+            in self.cleaned_data["imported_event"]
+        ):
+            events = events.filter(external_calendar__isnull=True)
+        elif (
+            calendar.EVENT_FROM_EXTERNAL_CALENDAR in self.cleaned_data["imported_event"]
+        ):
+            events = events.filter(external_calendar__isnull=False)
```

Comment:
That means the default behavior is that only non-imported events are shown, since both checkboxes are checkbed by default, but only the first case applies.

For me the expected behavior would be:
- Both checked: No filter applied
- One checked: Filter either for has external calendar or has no external calendar
- None checked: I guess this would be invalid and treated as both checked, like for the other checkboxes?

Reply:
For me this already works as you have described. Did it not work for you? :)

Reply:
What @david-venhoff meant is that filtering should be explicitly skipped by `pass` if both or none of the two choices are checked, like for the filter by recurrence type or `all_day`. 

The filter looks as it it were working fine, even without such explicit skipping, if you test by changing only the calender choices. To see it is actually needed, try the following steps:

1. Save https://github.com/user-attachments/files/17816539/outlook.calendar.txt as external calendar.
2. See 11 events in the event list (before applying any filter), all of them are recurring events.
3. Open the filter, uncheck "non-recurring events"
4. See only one event appears. (Expected: all events are shown)

Reply:
Thank you for explaining. This made it more clear to me! I hope I fixed what you meant now :) Could you please check again? :)